### PR TITLE
Add migration guide for lightning 2.0 upgrade

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/source/starthere/migration-guide.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+.. _dummy_header:
+
+* Replace ``trainer.strategy=null`` with ``trainer.strategy=auto`` as 
+  `lightning 2.0 doesn't have None strategy <https://lightning.ai/docs/pytorch/stable/common/trainer.html#:~:text=strategy%20(Union%5Bstr%2C%20Strategy%5D)%20%E2%80%93%20Supports%20different%20training%20strategies%20with%20aliases%20as%20well%20custom%20strategies.%20Default%3A%20%22auto%22.>`_.
+..
+* Remove ``resume_from_checkpoint`` if being used as a trainer flag and pass the path to 
+  `Trainer.fit(ckpt_path="...") method <https://lightning.ai/docs/pytorch/stable/upgrade/from_1_9.html#:~:text=used%20Trainer%E2%80%99s%20flag%20resume_from_checkpoint>`_.
+..
+* Set ``trainer.strategy = "ddp_find_unused_parameters_true"`` if there are unused parameters in your model as lightning 2.0 has find_unused_parameters as False by default.
```
Comment (ericharper) [inline docs/source/starthere/migration-guide.rst:17]: Is this the right syntax?

Comment (athitten) [reply docs/source/starthere/migration-guide.rst:17]: Yes it is: [we use it here](https://github.com/NVIDIA/NeMo/pull/6433/files#:~:text=Resolve%20conversation-,cfg.trainer.strategy%20%3D%20%22ddp_find_unused_parameters_true%22,-logging.info)

Comment (athitten) [reply docs/source/starthere/migration-guide.rst:17]: Also used in the lightning's PR: [strategy="ddp_find_unused_parameters_true"](https://github.com/Lightning-AI/lightning/pull/16611/files#:~:text=%60strategy%3D%22ddp_find_unused_parameters_true%22%60)

File: docs/source/starthere/migration-guide.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  is added as a precsion value for pure fp32 along with ``32``, ``'32'`` that existed. This can be taken into account as shown here in the `NeMo upgrade to lightning 2.0 PR <https://github.com/NVIDIA/NeMo/pull/6433/files#diff-e93ccae74f4b67d341676afc9f3c7e2c50f751ec64df84eb3b2a86b62029ef76R269>`_.
+..
+* Lightning 2.0 renames epoch end hooks from ``training_epoch_end``, ``validation_epoch_end``, ``test_epoch_end`` to ``on_train_epoch_end``, 
+  ``on_validation_epoch_end``, ``on_test_epoch_end``. The renamed hooks do not accept the outputs arg but instead outputs needs to be defined 
+  as an instance variable of the model class to which the outputs of the step needs to be manually appended. More detailed examples implementing 
+  this can be found under migration guide of `lightning's PR 16520 <https://github.com/Lightning-AI/lightning/pull/16520>`_. Example from NeMo 
+  can be found `here <https://github.com/NVIDIA/NeMo/pull/6433/files#diff-e93ccae74f4b67d341676afc9f3c7e2c50f751ec64df84eb3b2a86b62029ef76R904-R911>`_.
+..
+* Lightning 2.0 is not currently supporting multiple dataloders for validation and testing. The support for this will be added back soon in an
```
[the remark this answers is not in the document]
Comment (athitten) [reply docs/source/starthere/migration-guide.rst]: Added that.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**